dali/operators/math/expressions/arithmetic.cc

```diff
@@ -37,9 +36,9 @@ void ArithmeticGenericOp<CPUBackend>::RunImpl(HostWorkspace &ws) {
                                        {extent_idx, extent_idx + 1});
         }
       }
-    });
+    }, -task_idx);  // Descending numbers for FIFO execution
```

Comment:
Why do we want FIFO here?

Reply:
The work is already divided into similar sized chunks. Also @klecki is planning to change the work balancing so I'll leave it up to him to set the priorities here

Reply:
Sure. Can you extend comment there by this info?